### PR TITLE
cleanup: auto format markdown files using "mdformat"

### PR DESCRIPTION
**Pull request by devbww**
See https://mdformat.readthedocs.io/en/stable/users/style.html for a
description of the formatting style that `mdformat` follows.

`.mdformat.toml` specifies the tweakable formatting parameters, only
one of which we change from the default.  We set `number=true` to
minimize the diffs in the initial formatting, but it is probably best
to revert to the default `number=false` almost immediately.

Otherwise the most extensive edits are to:
- use `-` instead of `*` as the top-level bullet-list marker, and
- move all link definitions to the bottom of the document.

Fixes https://github.com/googleapis/google-cloud-cpp/issues/8470.

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/9721)

### CODE REVIEW

File: release/README.md [hunk 1]
```diff
@@ -153,20 +148,26 @@ take several weeks.
...
-* After review, publish the release.
-* Nudge `coryan@` to update our [vcpkg port].
+- After review, publish the release.
+- Nudge `coryan@` to update our [vcpkg port].
 
+[git-docs]: https://git-scm.com/doc
+[github-branch-settings]: https://github.com/googleapis/google-cloud-cpp/settings/branches
+[github-guides]: https://guides.github.com/
 [vcpkg port]: https://github.com/Microsoft/vcpkg/tree/master/ports/google-cloud-cpp
+[`changelog.md`]: /CHANGELOG.md
```
Comment (coryan) [inline release/README.md:172]: This did more than move the link, it also changed the case.  I am not sure how this renders?

File: CHANGELOG.md [hunk 2]
```diff
@@ -324,33 +306,33 @@ the APIs in these libraries are stable, and are ready for production use.
...
+- [DialogFlow CX](/google/cloud/dialogflow_cx/README.md)
+- [DialogFlow ES](/google/cloud/dialogflow_es/README.md)
 
 </details>
 
 ### [Bigtable](/google/cloud/bigtable/README.md)
 
-* feat(bigtable): add `AsyncWaitForConsistency()` helper for Table Admin ([#9310](https://github.com/googleapis/google-cloud-cpp/pull/9310))
-* doc: broken link in Bigtable README ([#9265](https://github.com/googleapis/google-cloud-cpp/pull/9265))
+- feat(bigtable): add `AsyncWaitForConsistency()` helper for Table Admin ([#9310](https://github.com/googleapis/google-cloud-cpp/pull/9310))
```
Comment (coryan) [inline CHANGELOG.md:317]: These are automatically generated as part of the release process.  Should we fix the script too?

File: google/cloud/workflows/quickstart/README.md [hunk 3]
```diff
@@ -5,21 +5,19 @@ client library in your own project. These instructions assume that you have
 some experience as a C++ developer and that you have a working C++ toolchain
 (compiler, linker, etc.) installed on your platform.
 
-* Packaging maintainers or developers who prefer to install the library in a
+- Packaging maintainers or developers who prefer to install the library in a
```
Comment (coryan) [inline google/cloud/workflows/quickstart/README.md:8]: Should we fix the generator to more closely follow this standard?

File: google/cloud/workflows/README.md [hunk 4]
```diff
@@ -34,6 +30,7 @@ to get started using this client library in a larger project. The following
 this library.
 
 <!-- inject-quickstart-start -->
+
```
Comment (coryan) [inline google/cloud/workflows/README.md:33]: The code here is generated by a script.  Should we fix the script too?

File: google/cloud/storage/quickstart/README.md [hunk 5]
```diff
@@ -160,21 +156,21 @@ Run the example, change the placeholder to appropriate values:
...
 bazel run :quickstart_grpc -- [BUCKET NAME]
- ```
+```
 
 ## Platform Specific Notes
 
 ### macOS
 
-gRPC [requires][grpc-roots-pem-bug] an environment variable to configure the
+gRPC \[requires\]\[grpc-roots-pem-bug\] an environment variable to configure the
```
Comment (coryan) [inline google/cloud/storage/quickstart/README.md:165]: This changes the rendering of `[requires][grpc-roots-pem-bug]` from a link to verbatim text.

File: google/cloud/spanner/benchmarks/README.md [hunk 6]
```diff
@@ -57,13 +55,10 @@ If you are running the benchmarks in a virtual machine, the library can
 automatically use the GCE instance service account (when you do **not** set
 `GOOGLE_APPLICATION_CREDENTIALS`). You may need to grant this service account
 permissions to work with Cloud Spanner. Examine the
-[spanner roles][spanner-roles-link] to chose a role for this account, the
+\[spanner roles\]\[spanner-roles-link\] to chose a role for this account, the
```
Comment (coryan) [inline google/cloud/spanner/benchmarks/README.md:58]: Another case of changing from a link to verbatim text

File: doc/packaging.md [hunk 7]
```diff
@@ -1,30 +1,25 @@
 # Packaging `google-cloud-cpp`
 
 <!-- This is an automatically generated file -->
+
```
Comment (coryan) [inline doc/packaging.md:4]: As the comment says, this is automatically generated. Maybe we should fix the script first and then reformat?

File: doc/contributor/howto-guide-update-googleapis-sha.md [hunk 8]
```diff
@@ -43,7 +43,7 @@ git commit -m"chore: update googleapis SHA circa $(date +%Y-%m-%d)" bazel cmake
 
 ```shell
 ci/cloudbuild/build.sh -t generate-libraries-pr
- ```
+```
```
Comment (coryan) [inline doc/contributor/howto-guide-update-googleapis-sha.md:46]: Nice fix.

File: doc/contributor/howto-guide-setup-cmake-environment.md [hunk 9]
```diff
@@ -32,7 +32,7 @@ cd google-cloud-cpp
 
 ## Download and bootstrap `vcpkg`
 
-[vcpkg] is a package manager for C++ that builds from source and installs any
+\[vcpkg\] is a package manager for C++ that builds from source and installs any
```
Comment (coryan) [inline doc/contributor/howto-guide-setup-cmake-environment.md:35]: This was broken already, there should be a link to `https://vcpkg.io` here.

File: release/README.md (hunk 1 above)
Comment (coryan) [reply release/README.md:172]: It renders fine, I checked in your branch.

File: google/cloud/storage/quickstart/README.md (hunk 5 above)
Comment (coryan) [reply google/cloud/storage/quickstart/README.md:165]: Well, no.  The link was broken already.

**Review by coryan — COMMENTED** [review]
This is too large for me to grok, and I fear I may miss some stuff.  Is there any way to break this into smaller changes?

Maybe by directory?  Maybe all the generated libraries first and then hand-crafted docs?

Or even do CHANGELOG.md it its own PR?

File: release/README.md (hunk 1 above)
Comment (devbww) [reply release/README.md:172]: Yes, the link tag is case independent, so the tool canonicalizes them.

File: google/cloud/storage/quickstart/README.md (hunk 5 above)
Comment (devbww) [reply google/cloud/storage/quickstart/README.md:165]: Right.  I plan to make a second pass, looking for escaped brackets as an indicator of something possibly wrong.

File: CHANGELOG.md (hunk 2 above)
Comment (devbww) [reply CHANGELOG.md:317]: In the bigger picture these fall into the same category (for me) as the generator producing code which is later fixed up by clang-format, and we shouldn't worry about it too much.  In the smaller picture, though, these are easier to fix, and I think we should do so as soon as someone notices and cares.

File: google/cloud/workflows/README.md (hunk 4 above)
Comment (devbww) [reply google/cloud/workflows/README.md:33]: Yes, eventually, I think.  See above.

File: google/cloud/workflows/quickstart/README.md (hunk 3 above)
Comment (devbww) [reply google/cloud/workflows/quickstart/README.md:8]: Yes, eventually, I think.  See above.

File: doc/packaging.md (hunk 7 above)
Comment (devbww) [reply doc/packaging.md:4]: I'm happy to come back and fix the script after, if you're happy with the opposite order.

File: google/cloud/spanner/benchmarks/README.md (hunk 6 above)
Comment (devbww) [reply google/cloud/spanner/benchmarks/README.md:58]: Another case of the link being broken before.  I'll come back to fix these.

**Review by devbww — COMMENTED** [review]
> This is too large for me to grok, and I fear I may miss some stuff. Is there any way to break this into smaller changes?

I could certainly reapply the changes in separate commits to make them a little easier to see. 
 Separate PRs would seem to require merging the changes before the automation, which I'm also happy to do.  (I definitely perused them in chunks.)

But it looks like you might have slogged through in any case.  It didn't take me too long to grok the general classes of things it was doing, and then being able to spot any special cases, and it appears that you've done the same.  I definitely gained confidence that the tool was always doing "the right thing".

So, just yell if you would prefer the more bottom-up approach where the "automate" step comes last and actually does nothing (too much).

File: doc/packaging.md (hunk 7 above)
Comment (coryan) [reply doc/packaging.md:4]: Fixing the script later SGTM.

File: google/cloud/bigtable/README.md [hunk 10]
```diff
@@ -80,22 +77,21 @@ int main(int argc, char* argv[]) try {
...
   fixed directory (such as `/usr/local` or `/opt`) should consult the
   [packaging guide](/doc/packaging.md).
-* Developers wanting to use this client library as part of a larger CMake or
+- Developers wanting to use this client library as part of a larger CMake or
   Bazel project should consult the aforementioned
   [quickstart](quickstart/README.md).
-* Developers wanting to compile the library just to run some of the examples or
-  tests should read [build and install](/README.md#build-and-install] section in
+- Developers wanting to compile the library just to run some of the examples or
+  tests should read \[build and install\](/README.md#build-and-install\] section in
```
Comment (coryan) [inline google/cloud/bigtable/README.md:90]: Another link that was already broken.

File: google/cloud/internal/README.md [hunk 11]
```diff
@@ -1,4 +1,4 @@
-# Implementation Details for gogle/cloud/*.{h,cc}
+# Implementation Details for gogle/cloud/\*.{h,cc}
```
Comment (coryan) [inline google/cloud/internal/README.md:1]: nice!

**Review by coryan — APPROVED** [review]
Okay.  I reviewed all the changes except for automatically generated README and quickstart/README files.

LGTM.

**Pull request merged**